packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,89 @@
         default:
           return 'navbar-dark bg-primary'
         }
-      }
+      },
     },
     methods: {
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      // add directory index, e.g http://foo/ or http://foo becomes http://foo/index.html
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url;
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`;
+        } else {
+          return `${url}/index.html`;
+        }
+      },
+      highlightLink(url) {
+        // Performs an exact equality match on links
+        if (this.hasLinkMatch(url, true)) {
+          return;
+        }
+        return this.hasLinkMatch(url, false);
+      },
+      hasLinkMatch(url, isStrict) {
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        for (const li of navLis) {
+          // there are two kinds of navlinks, get all
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const aNorm = this.normalizeUrl(a.href);
+            const urlNorm = this.normalizeUrl(url);
+            if (isStrict) {
+              // checks by strict equality of URL
+              if (aNorm === urlNorm) {
+                li.classList.add('current');
+                return true;
+              }
+            } else {
+              const aTagUrl = new URL(aNorm);
+              const browserUrl = new URL(urlNorm);
+              const aParts = `${aTagUrl.host}${aTagUrl.pathname}`.split('/').slice(1);
+              const bParts = `${browserUrl.host}${browserUrl.pathname}`.split('/').slice(1);
+              switch (this.defaultHighlightOn) {
```

Comment:
> In Navbar.vue, extract the data-highlight-on attribute if exact URL matching doesn't work

one more nit: the logic for this seems to be missing

could we shorten it to `highlight-on` as well? (for brevity on the author side)

Reply:
Will shorten the `highlight-on` part

@ang-zeyu I was under the impression that the user just needs to pass the `highlight-on` attribute like so: <navbar highlight-on='none'>, which should be a valid way to pass props to a vue component. After which, the prop is automatically extracted, so no explicit extraction should be needed. Is this correct?

Reply:
hmm to clarify, from your comment here https://github.com/MarkBind/markbind/pull/1379#issuecomment-723534533, the `highlight-on` configuration is an addition to option 2 discussed here https://github.com/MarkBind/markbind/pull/1379#discussion_r515606407 right?

my suggestion was to move the optional default configuration from your site.json suggestion to the navbar, so we can keep this entirely on the frontend, and allow different default behaviours for different navbars.

we would still need the ability to adjust the behaviour at a _link level_ (`<a highlight-on="...">`) because a single navbar can still require different sorts of highlighting (a simple example being the ug - exact for `home`, `showcase`, `about` but sibling-or-child for `user guide`).

> shorten

the shortening here refers to your proposal for `data-highlight-on` on `<a>` tags, not `default-highlight-on` (since this is just written once-per-navbar) as well

Reply:
Hmm I only had a navbar-level highlight-on configuration in mind, so there wasn't link-level highlighting